classes/Category.php

```diff
@@ -1527,13 +1527,12 @@ public function getParentsCategories($idLang = null)
 
         $categories = null;
         $idCurrent = $this->id;
-        if (count(Category::getCategoriesWithoutParent()) > 1
-            && Configuration::get('PS_MULTISHOP_FEATURE_ACTIVE')
-            && count(Shop::getShops(true, null, true)) !== 1) {
-            $context->shop->id_category = (int) Configuration::get('PS_ROOT_CATEGORY');
-        } elseif (!$context->shop->id) {
+        if (!$context->shop->id) {
             $context->shop = new Shop(Configuration::get('PS_SHOP_DEFAULT'));
         }
+        if (count(Category::getCategoriesWithoutParent()) > 1) {
```

Comment:
You removed the conditions `Configuration::get('PS_MULTISHOP_FEATURE_ACTIVE')` `count(Shop::getShops(true, null, true)) !== 1)`, why ?

If there's a good reason, then we need to ask QA to check the behavior with multishop

Reply:
because I think this test was buggy, testing two different things:
- wether you need to define a shop or not
- wether you need to use the root category or not

So I simplified the test:
- if there is no shop => instanciate one
- if there are several children in root category => use root instead of Home to display all categories

This feels more logical, maybe it won't respect the legacy behavior but the legacy behavior was incorrect anyway